### ISSUE TITLE
[FSR] Screenreaders do not receive alert message without setting focus on StreamlinedExplainer and ResolutionOptionsExplainer pages

### ISSUE THREAD

**Issue by josephrlee** [department-of-veterans-affairs/va.gov-team#69366]
## Description

During the AT study, we noticed when participants were using VoiceOver on a Mac and go to the Streamlined Waiver transition page, it wouldn't set focus on the heading in the `va-alert` component. I confirmed that this is an issue on the Resoultion Options explainer page as well. All the other pages have the h3 receive focus and auto-scroll, the interaction needs to be consistent on the alert pages as well.

[ResolutionExplainerWidget.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/7eb1a83a1bd53fcd0620c97d5251b133e2c99b18/src/applications/financial-status-report/components/resolution/ResolutionExplainerWidget.jsx)
[StreamlinedExplainer.jsx](https://github.com/department-of-veterans-affairs/vets-website/blob/430e1d59c61ca0fbf3c3572f4c8934c0d6aaa511/src/applications/financial-status-report/components/shared/StreamlinedExplainer.jsx)

Here is a related ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/52076 

Here is  a Loom of the issue: [Loom screen recording](https://www.loom.com/share/2f6a15891ff44864b8dd3ed314fec151)

Given: A screenreader user navigates the FSR
When: A screenreader user gets to the Resolution Options page or the Streamlined Explainer page.
Then:  They should have the `h3` in the `va-alert` component should receive focus be read aloud to them to maintain the flow of the previous pages.

## Tasks
- [x] Set focus to the `h3` in the `va-alert` component 

## Acceptance criteria

- [x] `h3` in the `va-alert` component receives focus on the `ResolutionExplainerWidget.jsx` and the `StreamlinedExplianer.jsx` pages.

## Testing
- [ ] N/A if non development work
- [ ] Testing passed and documented in this ticket based off the "Then" statement in the description

Testing procedure:
If additional testing steps or credentials to perform testing are needed list them here

---

## Ticket creation

<details><summary>Ticket requirements</summary>

### Required to be "Ready"
- [x] Description written
- [ ] Tasks defined
- [ ] Acceptance criteria written
- [ ] Testing defined
- [x] Add Labels
- [ ] Attach to an Epic
- [ ] Estimate Points
</details>

**Comment by josephrlee** [comment]
@Dr-Pongo, here's the initial ticket for the a11y bug we noticed during the AT study. I'll let you flush out the testing requirements and acceptance criteria and point it out. 

cc: @Tom-Davis @HeatherRienks @mgleo @charlottecesana  I've added it to the backlog for now

**Comment by josephrlee** [comment]
An alternative path to this if it's not feasible from a development standpoint. We could change the design of the pages to potentially move the `h3` out of the `va-alert`.

**Comment by HeatherRienks** [comment]
Per Refinement on 1/10/24 FE Team @Dr-Pongo @OptionSelect @amponce to take a quick look at this ticket. May need a Platform Support ticket?

**Comment by HeatherRienks** [comment]
We will plan for this to roll over to S117

**Comment by HeatherRienks** [comment]
FE is looking into this.

**Comment by HeatherRienks** [comment]
@Dr-Pongo will check with FE channel and then Platform to follow up on this issue. Stay tuned!

**Comment by HeatherRienks** [comment]
Will add to Sprint 118

**Comment by josephrlee** [comment]
I think I found the solution to fix this. Here's another [teams PR](https://github.com/department-of-veterans-affairs/vets-website/pull/23198). 

@Dr-Pongo Is this something we can attempt to do for our `va-alerts`?

**Comment by HeatherRienks** [comment]
@Dr-Pongo - any thoughts on this?

cc: @denisecoveyduc @Tom-Davis

**Comment by HeatherRienks** [comment]
I spoke with @Dr-Pongo and this  "might" be something that @kjsuarez could assist with. He would like to have a convo with Design to better understand the preference of an alert component there instead of using a more traditional page layout, but that part isn't a priority.

@josephrlee @mgleo @Tom-Davis @charlottecesana @denisecoveyduc

**Comment by amponce** [comment]
Resources:
https://github.com/department-of-veterans-affairs/vets-website/blob/e4ae4d9e5e9b093355a13406ccd7fab18347fbc6/src/platform/utilities/ui/focus.js#L43-L53

 * Discussion: https://dsva.slack.com/archives/CBU0KDSB1/p1676479946812439

**Comment by amponce** [comment]
PR with the fix is up - https://github.com/department-of-veterans-affairs/vets-website/pull/29700
<img width="681" alt="Screenshot 2024-05-09 at 1 28 46 PM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/3916436/b3c1b81a-f718-4442-b1a8-6f89d98c17b1">
